### ISSUE TITLE
Very inconsistant behaviour when deleting/uploading files using Zenodo API

### ISSUE THREAD

**Issue by serasset** [zenodo/zenodo#2502]
**Description of the problem:**
My use case : Using the API, I have to create a new version for a deposition, then remove all the (inherited) files, then upload a new set of files and bump the version number and publication_date.

I went through the full process, but most (BUT NOT ALL) deletion/upload requests generate a 403, Permission denied answer.

Hence, I had to retry several time before getting the whole process to be done.

**Full code that generated the error**

I just include the deletion code (one attempt) and the trace:

```python
import os
import os.path
import sys
import time
import copy
import glob
import datetime
from urllib.parse import urlparse
from random import randint
import requests, json

def attempt_purge(draft, token=ACCESS_TOKEN):
    '''Attempt to purge all files from the draft.
        returns the number of remaining files after the full deletion attempt
    '''
    # Get all file list of the draft
    draft_url = draft['links']['latest_draft']
    draft_contents = []
    while True:
        r_list = requests.get('%s/files' % (draft_url), params = {'access_token': token})
        if r_list.status_code == 200:
            draft_contents = json.loads(r_list.text)
            break
        else:
            time.sleep(25)
    if len(draft_contents) == 0:
        return 0
    else:
        deleted_files = 0
        print(f"Draft object has %d files." % len(draft_contents))
        #print(draft_contents)
        for file in draft_contents:
            time.sleep(randint(2, 5))
            r_delete = requests.delete('%s/files/%s' % (draft_url, file['id']), params = {'access_token': token})
            if r_delete.status_code == 204:
                print(f"DELETED %s" % ('%s/files/%s' % (draft_url, file['id'])))
                deleted_files += 1
            else:
                print('Error deleting existing file: ', file['id'], " / ", r_delete.text)
        print(f"Deleted %d files in the attempt" % deleted_files)
        return len(draft_contents) - deleted_files
```

Here I include the trace. You'll see that the number of files slowly decreases while some attempts are successful. 
```
Draft object has 8 files.
Error deleting existing file:  ff223769-35c4-443f-b13f-4c5f49b8be71  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2e9a920b-aa41-48ad-9b88-3d1ed0d3e8f4  /  {"status": 403, "message": "Permission denied."}
DELETED https://zenodo.org/api/deposit/depositions/10050201/files/e153a147-89a3-4320-992e-70f9a0e0cb08
Error deleting existing file:  73249ee7-e89f-4052-bcde-b940adfc56e7  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  c5053801-4013-4a10-9ac0-1304ad292d59  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2c9fa98a-9bf0-4847-a0e1-bc0e20c42883  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  585761c2-50ed-4a76-84fb-276cf578c89e  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  7eeaa83e-f4c1-4afc-8bde-52b7f71ce07c  /  {"status": 403, "message": "Permission denied."}
Deleted 1 files in the attempt
7 remaining files after attempt
Draft object has 7 files.
Error deleting existing file:  ff223769-35c4-443f-b13f-4c5f49b8be71  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2e9a920b-aa41-48ad-9b88-3d1ed0d3e8f4  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  73249ee7-e89f-4052-bcde-b940adfc56e7  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  c5053801-4013-4a10-9ac0-1304ad292d59  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2c9fa98a-9bf0-4847-a0e1-bc0e20c42883  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  585761c2-50ed-4a76-84fb-276cf578c89e  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  7eeaa83e-f4c1-4afc-8bde-52b7f71ce07c  /  {"status": 403, "message": "Permission denied."}
Deleted 0 files in the attempt
7 remaining files after attempt
Draft object has 7 files.
Error deleting existing file:  ff223769-35c4-443f-b13f-4c5f49b8be71  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2e9a920b-aa41-48ad-9b88-3d1ed0d3e8f4  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  73249ee7-e89f-4052-bcde-b940adfc56e7  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  c5053801-4013-4a10-9ac0-1304ad292d59  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2c9fa98a-9bf0-4847-a0e1-bc0e20c42883  /  {"status": 403, "message": "Permission denied."}
DELETED https://zenodo.org/api/deposit/depositions/10050201/files/585761c2-50ed-4a76-84fb-276cf578c89e
DELETED https://zenodo.org/api/deposit/depositions/10050201/files/7eeaa83e-f4c1-4afc-8bde-52b7f71ce07c
Deleted 2 files in the attempt
5 remaining files after attempt
Draft object has 5 files.
DELETED https://zenodo.org/api/deposit/depositions/10050201/files/ff223769-35c4-443f-b13f-4c5f49b8be71
Error deleting existing file:  2e9a920b-aa41-48ad-9b88-3d1ed0d3e8f4  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  73249ee7-e89f-4052-bcde-b940adfc56e7  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  c5053801-4013-4a10-9ac0-1304ad292d59  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2c9fa98a-9bf0-4847-a0e1-bc0e20c42883  /  {"status": 403, "message": "Permission denied."}
Deleted 1 files in the attempt
4 remaining files after attempt
Draft object has 4 files.
Error deleting existing file:  2e9a920b-aa41-48ad-9b88-3d1ed0d3e8f4  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  73249ee7-e89f-4052-bcde-b940adfc56e7  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  c5053801-4013-4a10-9ac0-1304ad292d59  /  {"status": 403, "message": "Permission denied."}
Error deleting existing file:  2c9fa98a-9bf0-4847-a0e1-bc0e20c42883  /  {"status": 403, "message": "Permission denied."}
Deleted 0 files in the attempt
4 remaining files after attempt
...
```

The same behaviour appear also for file upload (through new API with bucket).

**Expected behaviour**

All deletion should work at first attempt and I should not have to retry until everything is deleted.

**System information**

* Operating system: Mac OS X
* Virtual env with python 3.10, and jupyter
<details>
<summary>Output of pip freeze</summary>
<pre>

anyio==4.0.0
appnope==0.1.3
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.0
async-lru==2.0.4
attrs==23.1.0
Babel==2.13.1
backcall==0.2.0
beautifulsoup4==4.12.2
bleach==6.1.0
certifi==2023.7.22
cffi==1.16.0
charset-normalizer==3.3.1
comm==0.1.4
debugpy==1.8.0
decorator==5.1.1
defusedxml==0.7.1
exceptiongroup==1.1.3
executing==2.0.0
fastjsonschema==2.18.1
fqdn==1.5.1
idna==3.4
ipykernel==6.26.0
ipython==8.16.1
ipython-genutils==0.2.0
ipywidgets==8.1.1
isoduration==20.11.0
jedi==0.19.1
Jinja2==3.1.2
json5==0.9.14
jsonpointer==2.4
jsonschema==4.19.1
jsonschema-specifications==2023.7.1
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.8.0
jupyter-lsp==2.2.0
jupyter_client==8.5.0
jupyter_core==5.4.0
jupyter_server==2.9.1
jupyter_server_terminals==0.4.4
jupyterlab==4.0.7
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.9
jupyterlab_server==2.25.0
MarkupSafe==2.1.3
matplotlib-inline==0.1.6
mistune==3.0.2
nbclient==0.8.0
nbconvert==7.9.2
nbformat==5.9.2
nest-asyncio==1.5.8
notebook==7.0.6
notebook_shim==0.2.3
overrides==7.4.0
packaging==23.2
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
platformdirs==3.11.0
prometheus-client==0.17.1
prompt-toolkit==3.0.39
psutil==5.9.6
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
Pygments==2.16.1
python-dateutil==2.8.2
python-json-logger==2.0.7
PyYAML==6.0.1
pyzmq==25.1.1
qtconsole==5.4.4
QtPy==2.4.1
referencing==0.30.2
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.10.6
Send2Trash==1.8.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.5
stack-data==0.6.3
terminado==0.17.1
tinycss2==1.2.1
tomli==2.0.1
tornado==6.3.3
traitlets==5.12.0
types-python-dateutil==2.8.19.14
typing_extensions==4.8.0
uri-template==1.3.0
urllib3==2.0.7
wcwidth==0.2.8
webcolors==1.13
webencodings==0.5.1
websocket-client==1.6.4
widgetsnbextension==4.0.9

</pre>
</details>

**Comment by kikislater** [comment]
403, and SSL errors as well, I did the same like you in a while loop and sometimes upload needs 1, 2, ... 12 upload to succeed. SSL error break the while loop
It was more reliable before this Friday 13th :/ I wonder why this was not tested on sandbox url (the sandbox migration was backed to previous version due to non working stuffs) before having a sandbox on production. Strange strategy, we have promote the platform and we regret it now but there is no other alternative ...

**Comment by kikislater** [comment]
SSL error I mentionned visible here as well: 
https://github.com/lgloege/zenodopy/issues/14

**Comment by serasset** [comment]
I did not experiment SSL errors, but very unfrequent "bad gateway" answers.

**Comment by kikislater** [comment]
With curl I'm able to have some more: 

```
curl -v --upload-file /my/path/ortho.tif https://zenodo.org/api/files/xxxx-xxxxx-xxx--xxxx/ortho.tif?access_token=xxxxxxxxxxx

* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* We are completely uploaded and fine
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Mark bundle as not supporting multiuse
< HTTP/1.1 403 FORBIDDEN
< server: nginx
< date: Sun, 29 Oct 2023 17:59:00 GMT
< content-type: application/json
< content-length: 48
< access-control-allow-origin: *
< access-control-expose-headers: Content-Type, ETag, Link, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset
< set-cookie: csrftoken=..-; Expires=Sun, 05 Nov 2023 17:59:00 GMT; Max-Age=604800; Secure; Path=/; SameSite=Lax
< x-ratelimit-limit: 133
< x-ratelimit-remaining: 133
< x-ratelimit-reset: 1698602401.3410647
< retry-after: 60
< permissions-policy: interest-cohort=()
< x-frame-options: sameorigin
< x-xss-protection: 1; mode=block
< x-content-type-options: nosniff
< content-security-policy: default-src 'self' fonts.googleapis.com *.gstatic.com data: 'unsafe-inline' 'unsafe-eval' blob: zenodo-broker.web.cern.ch zenodo-broker-qa.web.cern.ch maxcdn.bootstrapcdn.com cdnjs.cloudflare.com ajax.googleapis.com webanalytics.web.cern.ch
< strict-transport-security: max-age=31556926; includeSubDomains
< referrer-policy: strict-origin-when-cross-origin
< set-cookie: session=.--; Secure; HttpOnly; Path=/; SameSite=Lax
< set-cookie: =; path=/; HttpOnly; Secure; SameSite=None
< 
* Connection #0 to host zenodo.org left intact
{"status": 403, "message": "Permission denied."}
```

`* We are completely uploaded and fine` and no data on zenodo ...  even with api or gui

**Comment by dshorthouse** [comment]
Related: https://github.com/zenodo/zenodo/issues/2493

Users I support are getting frustrated...

**Comment by serasset** [comment]
AFAICT, server has been unavailable (both on UI and API) for a small time November 1st around noon CET. 

Then everything worked like a charm. Deletions and uploads always succeeded until November 2, around 2pm CET. At that time, the UI worked (but file/draft deletion did not work). On API, I received 504 Gateway Time-out errors on file deletion while getting the draft/deposition object was working.

Then, the server worked again around November 2, 17:43. Seems that it was somehow restarted...

**Comment by kikislater** [comment]
I didn't try after 2pm CET today :/ 
Looks like another round of Friday the 13th nightmares ....

**Comment by Raen111** [comment]
И так будет всегда, постоянно ошибки ,с этим названием только
осоциироваться с сайтом ошибок

**Comment by dshorthouse** [comment]
> Then, the server worked again around November 2, 17:43. Seems that it was somehow restarted...

Everything seems to work again for me too, 7PM EDT.

**Comment by serasset** [comment]
Yes things seem to be working and stable again. Closing now
